docs/application/dotnet/guides/applications/component-based-app.md

```diff
@@ -0,0 +1,607 @@
+
+| Callback                               | Description                              |
+|----------------------------------------|------------------------------------------|
+| `OnLowMemoryCallback`                 | Event type for the callback function that is responsible for saving data from the main memory to a persistent memory or storage to avoid data loss. Data loss can occur if the Tizen platform low memory killer kills your application to free more memory. The callback function must release any cached data in the main memory to secure more free memory. |
+| `OnLowBatteryCallback`                | Event type for the callback function that is responsible for saving data from the main memory to a persistent memory or storage to avoid data loss in case of complete power failure. The callback function must also stop heavy CPU consumption or power consumption activities to save the remaining power. |
+| `OnDeviceOrientationChangedCallback` | Event type for the callback function that is responsible for changing the display orientation to match the device orientation. |
+| `OnLanguageChangedCallback`           | Event type for the callback function that is responsible for refreshing the display into a new language. |
+| `OnRegionFormatChangedCallback`      | Event type for the callback function that is responsible for refreshing the display into a new time zone. |
+| `OnSuspendedStateCallback`    | Event type for the callback function that is responsible for taking necessary actions before entering the suspended state or after exiting from the state. |
+
+
+
+<a name="service_component_states"></a>
+## Service Component States
+
+The following figure and table describe the service component states:
+
+![Running service component](./media/service_lifecycle.jpg)
+
+| State        | Description                         |
+|--------------|-------------------------------------|
+| `Create`      | The service component instance is created.            |
+| `Process Command`    | The service component instance handles the requested command. |
+| `Destroy` | The service component instance is destroyed.          |
+
+
+<a name="service_component_callbacks"></a>
+## Service Component Event Callbacks
+
+You can control the service component lifecycle by [managing the service component state events](#service_component_monitoring).
+
+The following table lists the callbacks for the service component state events:
+
+| Callback                     | Description                              |
+|------------------------------|------------------------------------------|
+| `OnCreate()`    | Used to take necessary actions before the lifecycle of a service component instance starts. Place the initialization code here. It will be called once in the instance's lifecycle. |
+| `OnRestoreContents()`    | Used to restore the current state of an instance. The data stored in  `OnSaveContent` will be passed by parameter. |
```

Comment:
Used to restore the current state of an instance. The data stored in `OnSaveContent` is passed by parameter.